controllers/secure.go

```diff
@@ -35,6 +35,34 @@ func (c *SecureContext) OAuth(rw web.ResponseWriter, req *web.Request, next web.
 	next(rw, req)
 }
 
+// LoginRequired is a middleware that requires a valid token or returns Unauthorized
+func (c *SecureContext) LoginRequired(rw web.ResponseWriter, r *web.Request, next web.NextMiddlewareFunc) {
+
+	// If there is no request just continue
+	if r == nil {
+		next(rw, r)
+		return
+	}
+
+	// Don't cache anything
+	// right now, there's a problem where when you initially logout and then
+	// revisit the server, you will get a bad view due to a caching issue.
+	// for now, we clear the cache for everything.
+	// TODO: revist and cache static assets.
+	rw.Header().Set("cache-control", "no-cache, no-store, must-revalidate, private")
```

Comment:
did you try removing this to verify it could fix the caching problem? (could leave that for another PR)